app/utils/poller.js

```diff
@@ -0,0 +1,29 @@
+import Ember from 'ember';
+
+export default Ember.Object.extend({
+  interval: function() {
+    return 10000; // Time between polls (in ms)
+  }.property().readOnly(),
+
+  // Schedules the function `f` to be executed every `interval` time.
+  schedule: function(f) {
+    return Ember.run.later(this, function() {
+      f.apply(this);
+      this.set('timer', this.schedule(f));
+    }, this.get('interval'));
+  },
+
+  // Stops the poller
+  stop: function() {
+    Ember.run.cancel(this.get('timer'));
+  },
+
+  // Starts the poller, i.e. executes the `onPoll` function every interval.
+  start: function() {
+    this.set('timer', this.schedule(this.get('onPoll')));
+  },
+
+  onPoll: function() {
+    // Issue JSON request and add data to the store
```

Comment:
Trigger an event (i.e. add Ember.Evented) that the caller can listen for instead of having one onPoll method?